tirocks-sys/build.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use cc::Build;
+use cmake::Config;
+use std::path::{Path, PathBuf};
+use std::{env, str};
+
+// On these platforms jemalloc-sys will use a prefixed jemalloc which cannot be linked together
+// with RocksDB.
+// See https://github.com/gnzlbg/jemallocator/blob/bfc89192971e026e6423d9ee5aaa02bc56585c58/jemalloc-sys/build.rs#L45
+const NO_JEMALLOC_TARGETS: &[&str] = &["android", "dragonfly", "musl", "darwin"];
+
+// Generate the bindings to rocksdb C-API.
+// Try to disable the generation of platform-related bindings.
+#[cfg(feature = "update-bindings")]
+fn bindgen_rocksdb(file_path: &Path) {
+    println!("cargo:rerun-if-env-changed=TEST_BIND");
+    let gen_tests = env::var("TEST_BIND").map_or(false, |s| s == "1");
+
+    let mut builder = bindgen::Builder::default();
+
+    if env::var("CARGO_CFG_TARGET_OS").map_or(false, |s| s == "windows") {
+        builder = builder.clang_arg("-D _WIN32_WINNT=0x600");
+    }
+    let builder = builder
+        .header("crocksdb/crocksdb/c.h")
+        .header("rocksdb/include/rocksdb/statistics.h")
+        .header("titan/include/titan/statistics.h")
+        .clang_arg("-xc++")
+        .clang_arg("-Irocksdb/include")
+        .clang_arg("-std=c++11")
+        .rustfmt_bindings(true)
+        .whitelist_function(r"\bcrocksdb_.*")
+        .whitelist_type(r"\bcrocksdb_.*")
+        .whitelist_var(r"\bcrocksdb_.*")
+        .whitelist_function(r"\bctitandb_.*")
+        .whitelist_type(r"\bctitandb_.*")
+        .whitelist_type(r"\brocksdb::Tickers")
+        .whitelist_type(r"\brocksdb::Histograms")
+        .whitelist_type(r"\brocksdb::titandb::TickerType")
+        .whitelist_type(r"\brocksdb::titandb::HistogramType")
+        .blacklist_type(r"\b__.*")
+        .derive_copy(false)
+        .opaque_type(r"\bctitandb_.*")
```

Comment:
why titan's type are opaque type

Reply:
Actually all structs are opaque types as the current way they are defined. But this is a stale change, I will remove it.